SLHCUpgradeSimulations/L1TrackTrigger/plugins/L1TkMuonFromExtendedProducer.cc

```diff
@@ -140,137 +179,188 @@ L1TkMuonFromExtendedProducer::produce(edm::Event& iEvent, const edm::EventSetup&
   for (auto l1mu : l1mus){
     imu++;
+
+    if( imu<MuonsMax ) {
+      MuonNum = imu + 1;
+      MuCn[imu].eta = l1mu.eta();
+      MuCn[imu].eta = l1mu.eta();
+      MuCn[imu].phi = l1mu.phi();
+      MuCn[imu].pt = l1mu.pt();
+      MuCn[imu].bx = l1mu.bx();
+      MuCn[imu].quality = l1mu.quality();
+      MuCn[imu].sigmaEta = l1mu.sigmaEta();
+      MuCn[imu].sigmaPhi = l1mu.sigmaPhi();
+      MuCn[imu].feta = fabs( l1mu.eta());
+  
+      const auto& gmtCand = l1mu.gmtMuonCand();
+      const auto& dtCand  = l1mu.dtCand();
+      const auto& cscCand = l1mu.cscCand();
+      const auto& rpcCand = l1mu.rpcCand();
+  
+      MuCn[imu].gmtCand = gmtCand.empty();
+      MuCn[imu].dtCand = dtCand.empty();
+      MuCn[imu].cscCand = cscCand.empty();
+      MuCn[imu].rpcCand = rpcCand.empty();
+  
+      LogDebug("L1TkMuonFromExtendedProducer") << " imu= " << imu
+        << " eta= " << MuCn[imu].eta 
+        << " phi= " << MuCn[imu].phi 
+        << " bx= " << MuCn[imu].bx
+        << " quality= " << MuCn[imu].quality
+        << " sigmaEta= " << MuCn[imu].sigmaEta
+        << " sigmaPhi= " << MuCn[imu].sigmaPhi 
+        << " gmtCand " << MuCn[imu].gmtCand 
+        << " dtCand " << MuCn[imu].dtCand 
+        << " cscCand " << MuCn[imu].cscCand 
+        << " rpcCand " << MuCn[imu].rpcCand;
+    } 
+    else { 
+      LogWarning("L1TkMuonFromExtendedProducer") << " Maximum Muon number too small " << imu;
+    };
+
+    // trCn filling
+  
+    PropState matchProp;
+    float drmin = 999;
+    int TrackNume = -1;
+    int il1tk = -1;
+    for (auto l1tk : l1tks ){
+      il1tk++;
+  
+      if( imu<MuonsMax ) {
+
+        unsigned int nPars = 4;
+        if (use5ParameterFit_) nPars = 5;
+        L1Tk[imu].nPars = nPars;
+        L1Tk[imu].pt = l1tk.getMomentum(nPars).perp();
+        if (L1Tk[imu].pt < PTMINTRA_) continue;
+  
+        L1Tk[imu].z  = l1tk.getPOCA(nPars).z();
+        if (fabs(L1Tk[imu].z) > ZMAX_) continue;
+  
+        L1Tk[imu].chi2 = l1tk.getChi2(nPars);
+        if (L1Tk[imu].chi2 > CHI2MAX_) continue;
+        
+        L1Tk[imu].nstubs = l1tk.getStubRefs().size();
+        if ( L1Tk[imu].nstubs < nStubsmin_) continue;
+  
+        L1Tk[imu].eta = l1tk.getMomentum(nPars).eta();
+        L1Tk[imu].phi = l1tk.getMomentum(nPars).phi();
+  
+        float dr2 = deltaR2(MuCn[imu].eta, MuCn[imu].phi, L1Tk[imu].eta, L1Tk[imu].phi);
+        if (dr2 > 0.3) continue;
+  
+        PropState pstate = propagateToGMT(l1tk);
+        if (!pstate.valid) continue;
+  
+        float dr2prop = deltaR2(MuCn[imu].eta, MuCn[imu].phi, pstate.eta, pstate.phi);
+
+        if (dr2prop < drmin){
+          drmin = dr2prop;
+          L1Tk[imu].idx = il1tk;
+          L1Tk[imu].matchPt = pstate.pt; 
+          L1Tk[imu].matchEta = pstate.eta;
+          L1Tk[imu].matchPhi = pstate.phi;
+          L1Tk[imu].matchSigmaEta = pstate.sigmaEta;
+          L1Tk[imu].matchSigmaPhi = pstate.sigmaPhi;
+        }
+        TrackNume = il1tk + 1;
```

Comment:
I don't see TrackNume used anywhere, I guess I'll have to wait to see when its use case is implemented